src/energyplus/ForwardTranslator/ForwardTranslateZoneMixing.cpp

```diff
@@ -50,21 +52,64 @@ namespace openstudio {
 namespace energyplus {
 
   boost::optional<IdfObject> ForwardTranslator::translateZoneMixing(ZoneMixing& modelObject) {
+
+    ModelObject zoneOrSpace = modelObject.zoneOrSpace();
+    boost::optional<ModelObject> sourceZoneOrSpace = modelObject.sourceZoneOrSpace();
+
+    auto getParentObjectName = [this](const ModelObject& mo) {
+      if (!m_excludeSpaceTranslation) {
+        return mo.nameString();
+      }
+
+      if (auto space_ = mo.optionalCast<Space>()) {
+        if (auto thermalZone_ = space_->thermalZone()) {
+          return thermalZone_->nameString();
+        } else {
+          OS_ASSERT(false);  // This shouldn't happen, since we removed all orphaned spaces earlier in the FT
+        }
+      }
+
+      return mo.nameString();
+    };
+
+    if (!sourceZoneOrSpace) {
+      if (m_excludeSpaceTranslation && modelObject.space()) {
+        LOG(Warn, modelObject.briefDescription()
+                    << " doesn't have a Source Zone or Space, it will not be translated. As you were using Space-Level ZoneMixing, and you are not "
+                       "translating to Spaces, it's possible it was pointing to two spaces inside the same zone");
+      } else {
+        LOG(Warn, modelObject.briefDescription() << " doesn't have a Source Zone or Space, it will not be translated.");
+      }
+      return boost::none;
+    }
+
+    if (zoneOrSpace == sourceZoneOrSpace.get()) {
+      // This isn't going to happen, because zm.setSourceSpace(newSpace) in ThermalZone::combineSpaces will be rejected
+      // Let's play it safe though
+      LOG(Warn, modelObject.briefDescription() << " has the same Receiving and Source Zone or Space, it will not be translated.");
+      if (!m_excludeSpaceTranslation) {
+        // We don't allow this at model time, the only reason we expect this to happen is when m_excludeSpaceTranslation is true, we call
+        // combineSpaces, and if the user has a ZoneMixing pointing to two spaces from the same ThermalZone, you end up with matching Receiving and
+        // Source Spaces
+        OS_ASSERT(false);
+      }
+      return boost::none;
+    }
```

Comment:
Ft adjust + warnings in case the user had a ZoneMixing pointing to two spaces **inside the same zone** AND **opting out of space translation**. Yep, that's the state of all the corner cases we have to test for nowadays...